OptimizelySDK.Tests/OptimizelyTest.cs

```diff
@@ -1642,5 +1642,52 @@ public void TestTrackListener(UserAttributes userAttributes, EventTags eventTags
         }
 
         #endregion // Test NotificationCenter
+
+        #region Test GetEnabledFeatures
+
+        [Test]
+        public void TestGetEnabledFeaturesWithInvalidDatafile()
+        {
+            var optly = new Optimizely("Random datafile", null, LoggerMock.Object);
+            Assert.IsEmpty(optly.GetEnabledFeatures("some_user", null));
+            
+            LoggerMock.Verify(l => l.Log(LogLevel.ERROR, "Datafile has invalid format. Failing 'GetEnabledFeatures'."), Times.Once);
+
+        }
+
+        [Test]
+        public void TestGetEnabledFeaturesWithNoFeatureEnabledForUser()
+        {
+            var userAttributes = new UserAttributes
+            {
+                { "device_type", "iPhone" },
+                { "location", "San Francisco" }
+            };
+
+            OptimizelyMock.Setup(om => om.IsFeatureEnabled(It.IsAny<string>(), TestUserId, It.IsAny<UserAttributes>())).Returns(false);
+            Assert.IsEmpty(OptimizelyMock.Object.GetEnabledFeatures(TestUserId, userAttributes));
+        }
+
+        [Test]
+        public void TestGetEnabledFeaturesWithAllFeatureEnabledForUser()
```

Comment:
Please be specific here, not all features should be returned. 
Mock some feature keys that should return false and shouldn't be added.